### ISSUE TITLE
Automatically perform conversion between GObject pointers

### ISSUE THREAD

**Issue by Matt5sean3** [Matt5sean3/GtkBuilderAid.jl#9]
The simplest way to do this would be to automatically add another function method for each function in the macro that takes a `Ptr{GObject}` as its first argument and uses ellipses to pass through to the actual function.

Something like:

``` julia
foo(obj::Ptr{GObject}, args...) = foo(GObject(obj), args...)
```

Also supporting this automatic conversion for userdata in certain cases would be nice. That's a bit more complicated because ellipses on non-final arguments aren't possible:

``` julia
foo(obj::Ptr{GObject}, args...) = if(isa(args[end], Ptr{GObject})
  foo(GObject(obj), args[1:end - 1]..., GObject(args[end]))
else
  foo(GObject(obj), args...)
end
```

The final issue is that this could clobber user function definitions, but I may just advise against creating definitions that result in such clobbering or add detection for such methods and skip .

**Comment by Matt5sean3** [comment]
Some forms of Gtk's `signal_connect` functions convert the GObject automatically, other forms don't. If they change that behavior it should just reach around whatever change I make here as multiple dispatch is sort of magic like that.

**Comment by Matt5sean3** [comment]
After trying the solution described above, that solution won't convert to a cfunction.

**Comment by Matt5sean3** [comment]
Next solution to attempt is to add the conversion to the start of the function.

**Comment by Matt5sean3** [comment]
On the `autoconvert` branch I've implemented this change. It only works on long form functions though.

**Comment by Matt5sean3** [comment]
Merged in the autoconvert branch #11
